### ISSUE TITLE
WinUI3 WebView2 mouse stop working while keyboard is still working.

### ISSUE THREAD

**Issue by applefanbois** [MicrosoftEdge/WebView2Feedback#3003]
There are 2 ways to replicate the problem.
It is new to the latest version stable version of WebView2.

-Click in an web page input box and type super fast.
    or
-Select text in a page and Ctrl + c multiple times super fast. 

The webview2 and the app will look like it is crashed and froze.
It is not frozen, keyboard is still working.
You can continue to type in the input box.
You can tab to change inputs.
You can even zoom with the keyboard.
CPU usage is zero.
The mouse does not work at all in the whole app, giving the illusion it is frozen.
So you can click in the webpage or the app part and it will do nothing.
I have not found a way to unstuck it.

The problem happens 5 times per hours for a normal user.
Typing fast is jus a way to make do the problem faster.
WebView2 is now unusable.

-Latest update of Windows 10 Pro version 22H2 build 19045.2311
-Also happening on Windows server 2022
-Microsoft.Windows.SDK.BuildTools 10.0.22621.755
-Microsoft.WindowsAppSDK 1.2.221116.1
-Microsoft Edge WebView2 Runtime 107.0.1418.42

--------------------- 2 cents
PS : If you are a developer and you are planning on using webview2 or any Microsoft SDK... don't
We already lost 200 000$ trying to make the MS SDK work and it never does.
Go with QT or anything else.
I wish I could go back in time and tell my younger self to stay away from any MS SDK.
If you do go with MS SDK, you will forever loose your sanity.
We have the exact same app with almost exact same code on iOS and macOS and never had any problems.

Pleas MS or anyone working on this : just make it work like QT, iOS, macOS, Android SDK.
I do not deserve this mental torture. My boss is so depressed that the app 4 years later is still not functional on Windows.
We are forced to ask our customers to buy Apple product to use our app.

**Comment by applefanbois** [comment]
If you type too fast in a web page input box, you select text and do many Ctrl-V or simply use the webview2 normally it will deactivate all mouse input in the whole app. A friend across the globe reported the exact same problem.

As of now, the webview2 is useless and so is our app.

Waiting for MS to fix it.


WebView2 disable all mouse inputs.
WebView2 freeze the whole app. (this is an illusion)

**Comment by ghord** [comment]
Hitting the same issue. Any updates?

**Comment by applefanbois** [comment]
We are now officially 3 people around the plant who reported this problem. The other person is a friend of min in another country. This is the scariest bug, you can't give an app with this problem to your customers.

I feel like I am crying for help in a desert. You cannot use WebView2 for more than 5 minutes without everything freezing. How can nobody noticed? because no one actually use it? The potential is there, WebView2 could be amazing, but we on the UWP/Win32/Reunion side are left alone while WPF get all the bell and whistles.

**Comment by ghord** [comment]
One thing I've noticed is that this only appears in WinUI3. When running Win32 WebView sample this issue doesn't appear. It seems to be connected with WinAppSdk 1.2. When I've tried older version (1.1.5) it does work without issues.

I've created new issue in WinAppSdk repository: https://github.com/microsoft/WindowsAppSDK/issues/3266

**Comment by applefanbois** [comment]
Indeed, thanks maybe it is related to WinUI3. 
I just hope they fix it as soon as possible.

**Comment by applefanbois** [comment]
The problem is still there in
Microsoft.WindowsAppSDK 1.2.221209.1

**Comment by Eilon** [comment]
Please refer to https://github.com/microsoft/WindowsAppSDK/issues/3266#issuecomment-1372636220 for a trivial WinUI3 + WebView2 repro.

**Comment by nathanjeynes** [comment]
This is affecting our blazor app in .net MAUI and means we cant release into production. This issue with the webview needs fixing asap.

**Comment by sebestyn168** [comment]
I left only one text field in my poc hoping it wont be rejected
Another issue I found with input date : open the calendar and move the window - the calendar stay still

**Comment by baaaaif** [comment]
we're affected too
--> upvote...

**Comment by NeilQ** [comment]
same issue and upvote...

**Comment by applefanbois** [comment]
WebView2 in its version 1.1 is so unstable, navigating make it crash, just reading text on a webpage makes it crash, plenty of memory violation... All the WebView2 are in a totally broken and useless state. MS managed to make Chromium unstable and broken.

Please fix it, our (your users) are suffering. Our customers on windows are suffering and they know the iOS/macOS version is working all the time. That is why we have more and more users buying Macs. 4 years later, we still don't have a working WebView on MS. Do I need to remind you that WEB is the #1 priority in a framework?

**Comment by Avenged** [comment]
I'm having the same issue. Any updates?

**Comment by applefanbois** [comment]
Just wait for another year or move your project to GTK, QT, Electron, pure Web if you can.
We made a bad decision to make an app on an abandoned OS with the official non-working framework.
Remember that MS just back stabbed everyone by killing UWP thus killing HoloLens.
You invest your hard work on Microsoft and you get punished.
Sadly we never learn, because we won't accept the situation : Windows is a platform to run legacy code only.

Even Microsoft is not using their own SDK/Framework. 
All their recent app are running in Electron.

Move your app to Android and it will run natively on Windows 11 through the Amazon app store.
Microsoft knows that its too late, so Windows invested a lot of ressources in running Android apps natively.

The future is only mobile and web. If you swim against the current, you will eventually drown.

Good luck.

**Comment by KieranDevvs** [comment]
How on earth has this not been addressed yet? How many likes and comments does an issue have before a team member even reads it? This is the problem with "feedback" repositories, we can't see what you're doing and it looks... well to put it bluntly... like you're doing absolutely nothing.

Hire a community support manager or make the project open source, because this system clearly just isn't cutting it...

**Comment by applefanbois** [comment]
Laugh, and the world laughs with you;. Weep, and you weep alone

**Comment by tofuandeve** [comment]
Hi all, thank you for reporting the issue. We are looking into this issue, however was unable to reproduce the issue with our internal WinUI3 test app. We will try with  the sample @Eilon  provided and update the status of the issue.
Thank you.

**Comment by tofuandeve** [comment]
I have tried with the sample code @Eilon provided and the issue still doesn't repro, the webview runtime version I am using is 109.0.1518.52.
Could you please try with that version of WebView2 runtime to see if you still see the issue @applefanbois ?

**Comment by baaaaif** [comment]
> I have tried with the sample code @Eilon provided and the issue still doesn't repro, the webview runtime version I am using is 109.0.1518.52. Could you please try with that version of WebView2 runtime to see if you still see the issue @applefanbois ?

haha, this was a good one... -.-
Hit ctrl and move the mouse for a few seconds. Breaks every webview2 in an WindowsAppSDK App >= v1.2, even with runtime 109.0.1518.52

**Comment by NeilQ** [comment]
> I have tried with the sample code @Eilon provided and the issue still doesn't repro, the webview runtime version I am using is 109.0.1518.52. Could you please try with that version of WebView2 runtime to see if you still see the issue @applefanbois ?

That caused by any quick text typing, so just copy some long text and paste(ctrl+v) to a text editor (like bing.com search box), paste, paste and paste, till the mouse frozen.

By the way it shows in any app which WindowsAppSDK >= v1.2 metioned above, v1.1.5 is fine. Not sure if caused by webview2 or WindowsAppSDK

**Comment by sebestyn168** [comment]
> That caused by any quick text typing, so just copy some long text and paste(ctrl+v) to a text editor (like bing.com search box), paste, paste and paste, till the mouse frozen.

I pasted 10k words of Lorem Ipsum and it is not systematic, sometimes the app manages to hold for a while and you can edit the text and click on other components, sometimes it freezes for 5 seconds before responding but manages to recover and some times it crashes right after pasting. Then the app crashes every time it is restarted when pasting again. Waiting before restarting seems to help.

**Comment by KieranDevvs** [comment]
> I have tried with the sample code @Eilon provided and the issue still doesn't repro, the webview runtime version I am using is 109.0.1518.52. Could you please try with that version of WebView2 runtime to see if you still see the issue @applefanbois ?

There are several issues linked to this from other repositories that have examples and reproduction steps.
Here is a video clearly showing the issue: https://www.youtube.com/watch?v=f5nYtta00y0

https://github.com/dotnet/maui/issues/12361

**Comment by DanTwomey** [comment]
This is a critical issue and is effecting our blazor app in .net MAUI and means we cant release into production. This needs fixing asap.

**Comment by lapulpeta** [comment]
This issue appears in our app but I haven't found consistent steps to reproduce it. It's very annoying.

**Comment by KieranDevvs** [comment]
>I have tried with the sample code @Eilon provided and the issue still doesn't repro, the webview runtime version I am using is 109.0.1518.52. Could you please try with that version of WebView2 runtime to see if you still see the issue @applefanbois ?

@danroth27 Can you please advise what runtime MAUI is using in the latest stable release?
All I can see is that MAUI is referencing:
```xml
<_MicrosoftWebWebView2Version>1.0.1462.37</_MicrosoftWebWebView2Version>
<PackageReference Update="Microsoft.Web.WebView2" Version="$(_MicrosoftWebWebView2Version)" />
```

**Comment by KieranDevvs** [comment]
Can confirm, this is still an issue with: `Microsoft.Maui.Sdk 7.0.58`

**Comment by Eilon** [comment]
@tofuandeve - I posted a standalone repro that uses only WinUI3 + WebView2, without any .NET MAUI: https://github.com/microsoft/WindowsAppSDK/issues/3266#issuecomment-1372636220

Ping me on Teams/email internally if you'd like to chat more about this.

**Comment by hartez** [comment]
I also run into this if I'm running the WinUI app on a VM via Remote Desktop and I switch to another VM. When I switch back, the mouse often (not always, but most of the time) stops working. Keyboard is still functional.

**Comment by pabloordonez** [comment]
We can reproduce this too within a WinUI 3 application (Windows App SDK 1.2.221209.1).
The two ways to reproduce, as mentioned here, is to write too much too fast, or to focus and blur the window. When navigating to other windows and getting back, sometimes the mouse stop responding. Also as mentioned, even if the mouse is blocked, keyboard continues to work, and you can navigate and operate the page normally. There's drop in performance, just the mouse getting disabled. 
There is no change in memory consumption, nor CPU usage. 
Is also worth mentioning that normally, after some time the mouse is unblocked. I was able to detect the source condition for the blockage, or to unblock it yet.

**Comment by tofuandeve** [comment]
Okay I got it to repro with @Eilon 's help! We have opened a bug for this with hi pri, and will start looking into fixing this.

Thank you all for all the helpful info to get the repro.

**Comment by pabloordonez** [comment]
Excellent news @tofuandeve !! Please let us know if you need something else! (And thank you!)

**Comment by KieranDevvs** [comment]
@tofuandeve Can we get updates seeing as the bug tracking isn't publicly accessible? Is there an ETA on this?

**Comment by tofuandeve** [comment]
Hi @KieranDevvs we are actively working on this issue with WinUI team right now as this issue is marked as highest priority of this sprint for us. 
We'll keep you posted on the progress, thank you for your patience.  

Thank you!

**Comment by koenvd** [comment]
We can also easily reproduce this when moving from a Parallels VM running the app to another screen outside the Parallels VM and then moving back to the app.

The app is built with WindowsAppSdk 1.2.221109.1, Webview2 Runtime 109.0.1518.70, running on Windows 11.  The issue does NOT reproduce when using WindowsAppSdk 1.4.  Both versions are targeting net6.0-windows10.0.19041.0. 

So pretty consistent with what has been shared here already but maybe it can still help 😀.

**Comment by tofuandeve** [comment]
Hi all, we are working with WinUI team and it seems like this is a regression from WindowAppsSDK, we passed the issue to the owner team and they will take a look at this asap. Will keep you posted on the progress!

Thanks!

**Comment by creacity** [comment]
I add a simple EditForm on Index.razor. As soon as I press the keyboard keys a little fast, the mouse is completely disabled.
(Microsoft.AspNetCore.Components.WebView.Maui 7.0.52)

```razor
@using System.ComponentModel.DataAnnotations
@page "/"

<h1>Hello, world!</h1>

Welcome to your new app.

 <EditForm Model="instance" OnSubmit="OnRegister">
    <DataAnnotationsValidator />
    <InputText @bind-Value="instance.CompanyName" />
    <InputText @bind-Value="instance.InstanceName" />
    <InputText @bind-Value="instance.ServerName" />
    <InputNumber @bind-Value="instance.Port" />
</EditForm>

<SurveyPrompt Title="How is Blazor working for you?" />

@code {
    protected Instance instance = new();

    protected override void OnInitialized()
    {

    }


    protected async Task OnRegister(EditContext form)
    {
        form.Validate();
        await Task.CompletedTask;
    }

    public class Instance
    {
        [Required]
        public string CompanyName { get; set; } = "";

        [Required]
        public string InstanceName { get; set; } = "";

        [Required]
        public string ServerName { get; set; } = "";

        [Range(1000, 9000)]
        public int Port { get; set; }
    }
}
```

**Comment by KieranDevvs** [comment]
> Hi all, we are working with WinUI team and it seems like this might be a regression from WindowAppsSDK, we passed the issue to the owner team and they will take a look at this asap. Will keep you posted on the progress!
> 
> Thanks!

Has an issue been opened within https://github.com/microsoft/WindowsAppSDK?

**Comment by nathanjeynes** [comment]
@tofuandeve Hi, could you provide an update as this has not received any updates for 2 weeks.  
Thanks

**Comment by Xyncgas** [comment]
Yea.. I wasn't crazy I was just trying to type and click and mess with my computer, I am a random person on a random machine but I found a way to 100% produce this issue which is basically what we are all saying here

**Comment by Xyncgas** [comment]
WPF / UWP / MAUI / Xamarin / WinForm / WinUI
Just stop Xamarin, don't bother shipping a bunch control, how about a cross platform GUI platform (like a browser) that doesn't use plaintext html or javascript engine that exposes its api for us to use directly in any language to compile the GUI solution with our functions calling its api, or indrectly by spinnin up exisitng GUI solution and talking to that process in binary, meanwhile as long as you let me put pixels on the screen control what each pixel looks like, provide abstractions like ok I am going to put a rectangle instead of just one pixel, let me say how many pixels the font size is or what's the color of this element, then finally what is user doing it them (clicked, rightclicked, mouse moved etc more features here) then I will feel enough

**Comment by gdar91** [comment]
This is a major blocker.

**Comment by KieranDevvs** [comment]
> This is a major blocker.

Clearly not major enough, judging by the way this is being dealt with.

**Comment by applefanbois** [comment]
WebView2 does not work since 9 months. Just creating a webview and loading the google page and type a few letter is enough to crash the app and MS does not care. There are no working web solution on windows right now.

MS has abandoned windows 25 years ago. Their solution is that Windows11 support natively Android app and they recommend using the Amazon android App Store. We should go make an android app and it will run better in windows than native app.

**Comment by tofuandeve** [comment]
> 
> Has an issue been opened within https://github.com/microsoft/WindowsAppSDK?
@KieranDevvs  Yes this issue is reported here  https://github.com/microsoft/WindowsAppSDK/issues/3266

**Comment by tofuandeve** [comment]
As of today, WinUI team is actively working on investigating this issue. I will let you know as soon as we get more updates from them. Thank you for your patience.

Thanks!

**Comment by true-perfect-code** [comment]
I am sometimes very angry at Microsoft myself, even though I earn my money professionally thanks to Microsoft programming. But then I always remind myself that the people who develop Microsoft products feel the same way I do when I unknowingly introduce a bug somewhere in my software. So we should remain proper compared to our colleagues (from Microsoft), because what these different Microsoft teams bring together mostly exceeds everything that every programmer of us has done in his life! at this point a big respect to all who develop Microsoft products together.

I want to get rid of something else. If you restrict yourself to Windows only (e.g. winForms or WPF), then you have no problems. I have developed a lot of Windows applications in the last 20 years and never had any problems with libraries, frameworks or the like.
But it's a different story if you want to develop for other platforms, and I think that all this anger towards Microsoft comes from the fact that Microsoft still can't manage to provide a framework that you can program with fully support to other platforms.
That actually comes from the fact that Microsoft has always tried to rely on their own solutions. But this has changed a lot in the last years, especially with Blazor. Unfortunately it takes a lot of time until all Microsoft ideas have matured (e.g. MAUI Blazor).
My personal opinion is that especially MAUI Blazor could be the ultimate development platform. But I was so desperate and disappointed because of so many bugs, that I decided to switch to Flutter last year, so that I can develop programs with one code base for all systems. After about 2 months of intensive training and 2 implemented projects I went back to MAU Blazor. I don't want to go into the reasons here, that would take too long. 

But my conclusion is that we have to be patient, because the way Microsoft is going, Google did it with Flutter and others too. Once MAUI Blazor is stable, we will benefit a lot from it and we will be able to develop apps for all platforms (Windows, Web, Android or iOS). Don't be surprised about the web listing, officially MAUI only supports native app development, but if you set up a Visual Studio project cleverly using Blazor, then web is also possible with the same code.

I'm very happy with Blazor (and now MAUI), but I'm annoyed by all the bugs and I can only hope that we'll get a stable MAUI (Blazor) at least this year.

Big thanks to Microsoft Teams

**Comment by applefanbois** [comment]
The fact that there are many ways to make an app for windows is the source of all the problems.
There is only 1 way to make an app on macOS and iOS, there is only 1 language and there is only 1 sdk.
Android is the same, only 1 framework/sdk and only 1 language.

There are 20 official way to make app for windows (15 of them are already deprecated and abandoned). This is too much and it will never work.

**Comment by KieranDevvs** [comment]
> I am sometimes very angry at Microsoft myself, even though I earn my money professionally thanks to Microsoft programming. But then I always remind myself that the people who develop Microsoft products feel the same way I do when I unknowingly introduce a bug somewhere in my software. So we should remain proper compared to our colleagues (from Microsoft), because what these different Microsoft teams bring together mostly exceeds everything that every programmer of us has done in his life! at this point a big respect to all who develop Microsoft products together.
> 
> I want to get rid of something else. If you restrict yourself to Windows only (e.g. winForms or WPF), then you have no problems. I have developed a lot of Windows applications in the last 20 years and never had any problems with libraries, frameworks or the like. But it's a different story if you want to develop for other platforms, and I think that all this anger towards Microsoft comes from the fact that Microsoft still can't manage to provide a framework that you can program with fully support to other platforms. That actually comes from the fact that Microsoft has always tried to rely on their own solutions. But this has changed a lot in the last years, especially with Blazor. Unfortunately it takes a lot of time until all Microsoft ideas have matured (e.g. MAUI Blazor). My personal opinion is that especially MAUI Blazor could be the ultimate development platform. But I was so desperate and disappointed because of so many bugs, that I decided to switch to Flutter last year, so that I can develop programs with one code base for all systems. After about 2 months of intensive training and 2 implemented projects I went back to MAU Blazor. I don't want to go into the reasons here, that would take too long.
> 
> But my conclusion is that we have to be patient, because the way Microsoft is going, Google did it with Flutter and others too. Once MAUI Blazor is stable, we will benefit a lot from it and we will be able to develop apps for all platforms (Windows, Web, Android or iOS). Don't be surprised about the web listing, officially MAUI only supports native app development, but if you set up a Visual Studio project cleverly using Blazor, then web is also possible with the same code.
> 
> I'm very happy with Blazor (and now MAUI), but I'm annoyed by all the bugs and I can only hope that we'll get a stable MAUI (Blazor) at least this year.
> 
> Big thanks to Microsoft Teams

I'm sorry but no. You don't get to pull the "we're human, we make mistakes" card. Making a mistake would be: the bug occurring, being filed as a P1 issue, a staff member promptly picking the issue up, and delivering a fix within a reasonable time frame in accordance with how complicated the issue is. And if the issue is complicated thus will take some time to resolve / is being blocked by something else, then clear and frequent updates would be given to users so they can make informed actions as to how they want to deal with their own projects.

What actually happened was several people raised the same bug in different posts in **November** last year after the release of .NET 7, and the issue wasn't even looked at until several months after the initial post. And that was only because other teams got an influx of problems relating to this dependency, and no one was doing anything about it. Even to this day, the problem isn't fixed, there's no ETA on a resolution, and its been passed off to another team whos tracking it internally, and isn't giving updates to any of its users *yet again*.

**Comment by oliverw** [comment]
> There are 20 official way to make app for windows (15 of them are already deprecated and abandoned). This is too much and it will never work.

It's not like someone forced Microsoft to go that route. This is entirely on them. The really, really baffling and concerning part is that high profile issues - such as this one - do not appear to be of particular importance internally. I guess that's what you get when you are not dogfooding your own stuff like Apple and Google do (not that I prefer their technologies over .NET, but at least they stick to their guns).

**Comment by creacity** [comment]
@tofuandeve We are nearing completion of our product and are completely bogged down by this issue. Please provide an ETA or workaround/patch for this issue. We are in a desperate situation.

**Comment by applefanbois** [comment]
I am gonna be real here. MS failed to give us a proper WebView for 10 years, so you can probably wait another 5 years. And when that thing is stable, they will cancel it like they did with UWP and HoloLens.

It would be wiser to start over again with QT, GTK or Electron.

Another best solution would be to make an Android App and put it on the Android Amazon Store because it would run natively on Windows 11. Sadly, Windows Server are not compatible with Windows (Windows Server has no MS Store, NO Amazon Android App Store and does not support native android app). MS Windows died 20 years ago and you just refuse to acknowledge reality. Don't hurt yourself, it is time to move on.

**Comment by alelom** [comment]
> @tofuandeve We are nearing completion of our product and are completely bogged down by this issue. Please provide an ETA or workaround/patch for this issue. We are in a desperate situation.

Just reporting that I am in the exact same situation. This bug costed us hours of research to find a workaround which cannot be done. It is a major dealbreaker for our UI developed in MAUI Hybrid.

**Comment by lapulpeta** [comment]
The lack of updates is troubling to say the least

**Comment by sebestyn168** [comment]
Yes, I'm afraid MS laying off 10,000 people isn't helping. 
As for Apple's Fan Boy, it would be nice if he went and made his app on Android and stopped busting everyone's balls. Doomsayers are just good at making bad situations worse and hoping they can say they were right to add their shovel of dirt.

**Comment by tofuandeve** [comment]
Hi all, this bug is set as top priority issue right now for owner team and as of today the owner team has narrowed down the root cause and they are working on a fix for this and setting a goal to get it fix as soon as possible.

Thanks!

**Comment by Xyncgas** [comment]
I remember webview worked before, because I have an application built with MAUI.Blazor that works, if you are fixing a problem that's blocking everyone maybe provide alternative solutions for example how to switch to the webview that works for my MAUI.Blazor project or maybe release a hot fix for Visual studio to use it for now

**Comment by tofuandeve** [comment]
@Xyncgas I can check with WinUI team to see if there is any workaround they recommend, but for now from Webview2 perspective, switching WindowsAppSDK back to version before v1.2 will make it work again.

**Comment by true-perfect-code** [comment]
What should be much more important for Microsoft and all of us is management of bugs. The most crucial question I always ask myself is, how can it be that a lot of time passes until the teams realize that the bug is theirs?

This bug was reported on Nov 15! On 21 Dec, someone complains that there are only 3 people on the planet who have just reported this problem (probably because there are really very few of us who use MAUI Blazor and therefore rely on WebView. XAML does not have this problem).

If I'm not wrong, someone from Microsoft (@Eilon) reported bug on 5 jan!? Then it became a bit more concrete on 18 Jan with @tofuandeve report.

And only 5 days ago (Feb 18) the rigth MS Team started to fix the problem!

If the right team had received this error message immediately, the problem would probably not have been noticed at all, because he would have been fixed immediately with the next update. but so we have lost two months and the error is still in there!

**Respect to microsoft, but at this point Microsoft should think about how to improve quality management.**

**Comment by BieleckiLtd** [comment]
Oh my world. I was hoping to find what I'm doing wrong, but not this. It's WebView2 major major issue. It prevents us shipping Blazor Hybrid app no matter if it is WPF, winforms, MAUI... we're stuck. Please prioritise. Thanks.

**Comment by applefanbois** [comment]
This is an unacceptable level of testing. 
We now know that the people making WebView2 do not even use it.
They did not even bother to load any web page and type something in a search engine or anywhere else.

Just creating a WebView2 and typing crash everything within 10 seconds of normal usage.
MS did not even bother to even try it.
We also know that MS are not using their own stuff, they prefer electron.

And the list of important features missing are huge : like extracting the HTTP code (404 for example).
I have over 30 bugs and important limitations that I have provided to our shareholder.

I estimate that the cost of making an app with MS stuff is 10x bigger than Android, Linux or Apple.
We lost 300 000$, the app was completed 3 years ago, I am still working hard to make it work.

Every MS framework call can and will crash the entire app at any time (even inside a try catch block).
Nothing is working, absolutely nothing : sliders are not working, events are not working, buttons are not working, multi-threading is always re-entering the same code before it complet on the main thread....

We even found a bug yesterday where if somebody change their Windows Server language, the app will start missing words here and there at random. Deleting the app settings (create by MS, not me) solve the problem. There is not a single day without one of our customer reporting a bug or a crash (not related to my code).

And I am not even talking the performance, everything is 400x slower than android and apple (tested with the same code).
MS is so slow, I had to disable animation and sound. It took me 3 months to be able to memorize window position and size with display scaling because MS removed the feature. In fact I am the only one on the internet who managed to make it work in all circonstances. 

MS Frameworks are development hell. My employer is a huge Windows fan and he is used to stuff never working, he suffer like all Windows user from Stockholm syndrome where the unacceptable becomes normal. People on other OS don't have crashes every day, on other OS stuff is always working by default.

As a Linux, Android and Apple user, I am appalled by the way Windows behave, I am not used to stuff not working. I tasted the sugary fruit and I can't stand this rotten sour thing that is Windows and native development on Windows.

Do you think it is normal for a windows port to only have 1/10 of the features of the Android, iOS and macOS version? MS does not even offer any dev tools to deal with PDF (like Linux, Android and Apple does). Windows in 2023 still has no screen recorder??? What is this, the 90's?

**Comment by oliverw** [comment]
@applefanbois Dude, seriously. Knock it off. The situation is bad, but posting rant after rant at a place that is most likely not even read by the team responsible for WebView is just wasting everybody's time.

**Comment by applefanbois** [comment]
I don't want other people to suffer as much as we did/are.

I want future generations to remember how MS made our lives miserable.

**Comment by Xyncgas** [comment]
> @applefanbois Dude, seriously. Knock it off.
It's ok

**Comment by gdar91** [comment]
@BieleckiLtd It works in WPF. We migrated our app from maui to wpf VebView and haven’t had any freezes yet.

**Comment by true-perfect-code** [comment]
@applefanbois
Unfortunately I have to disagree! I work with Windows, almost since Windows exists and also know the DOS time. But it has never been as bad as you describe it here.

I have done a dozen large projects for corporations, starting with C, C++ (MFC) and ending with .NET and C# from 2002 (and in some custom projects even with VB.NET). Always on Windows platform, but never had any of the problems you describe.

The question is also how do you program? Without you taking it badly, I've come across several programming projects so far where I've wondered if it's really possible to program in such a complicated and unusable way.

And lastly, I have to say that I'm quite surprised about your post because I've never read anything like this before and believe me, I've been in the business for decades. If it was so bad with Microsoft, I would have noticed it myself until now!

**Comment by applefanbois** [comment]
My app is not just a small calculator project, but the app is almost as complex as a 3D software like Blender.
I have almost 1 million lines of code and zero code repetition, every call to framework and os fully wrapped.
My app use many complex things like web context isolation, multi viewports and multi language (text, voice, speech) at the same time... And nothing is blocking.

I could copy/paste here everything that does not work with WinUI3 and all of them are basic things.
The basic stuff do not work, sliders don't trigger onRelease events for example.
Actually, they did but that was 2 years ago.

And most stuff that do work are so low in quality that I removed the feature.

Like I said, because you are used to being disrespected on Windows, you cannot grasp how things can truly be amazing on Linux, Android and Apple. Apple product are more expensive, but even at lower specs, they run faster because their code is better. For example unzipping a 1gig file on windows can take up to 5 minutes while on macOS it is 5 seconds. You would not know that because you did not test it. I am a multi-platform dev, so I not only experience these things, I can mesure it. My app on iOS/macOS is 16 meg, the exact same app on Windows is 96 megs.

**Comment by sebestyn168** [comment]
@true-perfect-code Don't bother, he just wants to brag about how cool he is - he just forgets to say how he would have done everything better than MS with his unlimited resources

**Comment by applefanbois** [comment]
You think it is normal for a webview to not support any typing from the keyboard?
Typical abnormal windows user reaction.

Is it too much to ask for things to work like they do on Android and Apple?

MS made an event for onSliderReleased, but it does not work? 
Why doing stuff that does not work is beyond me.

It is not liking or taste here. Can you imagine buying a car that does not start, wipers that do not work when it rains, brakes that sometimes accelerate the car randomly or a steering wheel that only turn left? This is what MS is doing. Nothing to do with me, these things have been reported. I could past 30 links of major problems with WinUI3.

WinUI3 multiple windows is not working or way too unstable to implement.
WinUI3 camera does not work.
WinUi3 voice and speech do not work.
WinUI3 cannot print without dialog out of your control, so it is impossible to make a ticket distributer or invoice printing app.

Forums are packed with missing and broken feature. 
You never see that elsewhere.

But the problem is that Microsoft told us what was their priority and it was not related to technology.
https://www.youtube.com/watch?v=ih2RLpJ1ikU

**Comment by Kanawanagasaki** [comment]
Can you guys please stop? I subscribed to this thread in order to get notifications about the current status of the bug. I do not care about anyone's experience with Windows or Microsoft, do not care what technology is better, do not care what you think Microsoft should do. Here, in this thread, I only care about the status of the bug. If you want to discuss something except the issue's theme please choose different platform/thread/way for communication. Thank you.

**Comment by Xyncgas** [comment]
> I subscribed to this thread in order to get...
> If you want to discuss something except the issue's theme please choose different...

I have a very nice email client that shows me which group of email I am getting using AI sorting, I can select keywords like who's replying in the issue to highlight the email. besides from finding out when the bug gets fixed, I am also interested in what other people are experiencing, as we are helping each other to form opinions in an open discourse to the issues we are facing and make the choices according to the information. Censoring people is like removing the pricing tag from product, it's simply unacceptable.

**Comment by Stormwalker** [comment]
I don't think this is classed as "censorship" - the poster here has made his point that he dislikes Windows quite well - I mean... we get it.  No need to keep banging on and on and on about it. We know. So unless there's something new and useful to say on the thread topic then what's the point?  Nothing has been deleted, nothing has been censored.  Just... we get it.

**Comment by Xyncgas** [comment]
yeah I am tired of hearing how earth is sick and how african is dying too, but we share a world together in exchange of listening to others complaining, I would rather get to talk to people than being a nerd, my mom nags me all the time, my boss keeps saying weird stuffs to me, but I love this world. You see, love is what made us together, writing program, talking to AI. Yes sometimes it goes a little out of hand, we create something terrible before we knew it, but because of we get to complain to each other, make fun at each other, esteem each other, we are together

To put it in president Xi's word, we are a single community with our fate binded together

There is a red line that we can not cross, on the other side there is ignorance, fear, censorship, racism, on this side we have tolerance. In order to help create a global community, where everyone's opinion is just as important as technological facts, we shouldn't cross this line imo

**Comment by true-perfect-code** [comment]
I deleted all my posts because @Kanawanagasaki is right. You should really just focus on the problem here.

I suggest that everyone else also delete their posts that are not related to the topic. After all, we're all here to get info about the WebView bug, not discuss Microsoft.

(I will delete this post in the end of the day as well)

**Comment by alengYuan** [comment]
It seems that the problem would not occur until I set the following values:
```C#
WebView.CoreWebView2.Settings.AreBrowserAcceleratorKeysEnabled = false;
WebView.CoreWebView2.Settings.AreDefaultContextMenusEnabled = false;
WebView.CoreWebView2.Settings.AreDevToolsEnabled = false;
WebView.CoreWebView2.Settings.IsGeneralAutofillEnabled = false;
WebView.CoreWebView2.Settings.IsPasswordAutosaveEnabled = false;
WebView.CoreWebView2.Settings.IsPinchZoomEnabled = false;
WebView.CoreWebView2.Settings.IsStatusBarEnabled = false;
WebView.CoreWebView2.Settings.IsSwipeNavigationEnabled = false;
WebView.CoreWebView2.Settings.IsZoomControlEnabled = false;
```

**Comment by orosbogdan** [comment]
Same issue on .NET MAUI (.net 7).

**Comment by Day-Go** [comment]
> Hi all, this bug is set as top priority issue right now for owner team and as of today the owner team has narrowed down the root cause and they are working on a fix for this and setting a goal to get it fix as soon as possible.
> 
> Thanks!

Any updates on the status of the fix?

**Comment by Xyncgas** [comment]
> I left only one text field in my poc hoping it wont be rejected

If you have only one text field this can still happen

**Comment by Xyncgas** [comment]
is @tofuandeve still the guy assigned to this, is this guy the only person who's working on this

**Comment by KieranDevvs** [comment]
> is @tofuandeve still the guy assigned to this, is this guy the only person who's working on this

Its been assigned to another team and is being tracked here: https://github.com/microsoft/WindowsAppSDK/issues/3266
But don't expect to get any updates. I've been trying for weeks.

**Comment by Xyncgas** [comment]
It's fixed allegedly

**Comment by alelom** [comment]
> It's fixed allegedly

Source?

**Comment by creacity** [comment]
@alelom https://github.com/microsoft/WindowsAppSDK/issues/3266#event-8632023034

**Comment by true-perfect-code** [comment]
@tofuandeve
Any news about an update possibility?

**Comment by applefanbois** [comment]
no

**Comment by koenvd** [comment]
Just to make everyone aware but a fixed is being tested it seems https://github.com/microsoft/WindowsAppSDK/issues/3266#issuecomment-1458501446 🤞

**Comment by applefanbois** [comment]
Good News

**Comment by DM-98** [comment]
I had same issue in Blazor Hybrid MAUI ([https://www.youtube.com/watch?v=f5nYtta00y0]), and when I removed OnAfterRenderAsync, it started working smoothly again. Logic had StateHasChanged();, which I think was the issue.

**Comment by KieranDevvs** [comment]
> I had same issue in Blazor Hybrid MAUI [(video demonstration)](https://www.youtube.com/watch?v=f5nYtta00y0) and when I removed OnAfterRenderAsync, it started working smoothly again. Logic had StateHasChanged();, which I think was the issue.

That's my example video and the issue has nothing to do with Blazor thus "removing OnAfterRenderAsync" wouldn't do anything to prevent the problem. The cause has been fixed with a new version of WinAppSDK which will come to MAUI Blazor when the MAUI team upgrade the template version, or you can force it by manually specifying the NuGet package reference in the CSProj file.

**Comment by DM-98** [comment]
> > I had same issue in Blazor Hybrid MAUI [(video demonstration)](https://www.youtube.com/watch?v=f5nYtta00y0) and when I removed OnAfterRenderAsync, it started working smoothly again. Logic had StateHasChanged();, which I think was the issue.
> 
> That's my example video and the issue has nothing to do with Blazor thus "removing OnAfterRenderAsync" wouldn't do anything to prevent the problem. The cause has been fixed with a new version of WinAppSDK which will come to MAUI Blazor when the MAUI team upgrade the template version, or you can force it by manually specifying the NuGet package reference in the CSProj file.

I saw the video, it applied to me as well, which is why I referenced your video. The issue was in the Blazor which I used. And removing the OnAfterRenderAsync on my mainlayout did fix my issue I had for a while. That's only good if they've fixed it in the next release.

**Comment by applefanbois** [comment]
The latest update is working fine for now.

Tell me, nobody noticed that the file drag & drop does not work in WebView2? It has never work, even when reported 2 years ago by many people. Another day, another basic feature not working.

1 problem fixed, now go fixe the remaining 100 000 broken and missing features in WebView2.

**Comment by c-npolyak** [comment]
I noticed the same problem with WinUI3.

**Comment by c-npolyak** [comment]
> 



> The latest update is working fine for now.
> 
> Tell me, nobody noticed that the file drag & drop does not work in WebView2? It has never work, even when reported 2 years ago by many people. Another day, another basic feature not working.
> 
> 1 problem fixed, now go fixe the remaining 100 000 broken and missing features in WebView2.

@applefanbois which update is working for you? I thought I had a newest WinUI and it is still  not working.

**Comment by applefanbois** [comment]
How is it possible that the file drag and drop is not working in a modern web engine? Do I really need to report it? Am I the only one using it? Maybe MS should have used the Safari WKWebView engine instead? WebView2 is using the Chrome engine. Is the chrome engine so limited and unreliable out of the box? 2 years and it lacks all the basic features expected from a webview and most stuff that is there is not working correctly.

**Comment by c-npolyak** [comment]
> How is it possible that the file drag and drop is not working in a modern web engine? Do I really need to report it? Am I the only one using it? Maybe MS should have used the Safari WKWebView engine instead? WebView2 is using the Chrome engine. Is the chrome engine so limited and unreliable out of the box? 2 years and it lacks all the basic features expected from a webview and most stuff that is there is not working correctly.

After MS killed Silverlight - a great product from my point of view, I am very cautious about MS UI packages. Though I still consider C# the best existing language for development (especially now after it became multiplatform and open source).

**Comment by applefanbois** [comment]
There is ASM, C and all the other cheap programming languages that are basically C for dummies. C# is ok, but nothing more than a c++ that generate bytecode instead of Assembly.

**Comment by koenvd** [comment]
Can you close this issue as resolved now @applefanbois or @tofuandeve ?
Thanks!

**Comment by applefanbois** [comment]
The problem is still there.

**Comment by Xyncgas** [comment]
> The problem is still there.

If you create a new MAUI.Blazor project does this problem still exist

**Comment by danroth27** [comment]
> If you create a new MAUI.Blazor project does this problem still exist

To get the fix in a .NET MAUI app, please upgrade the Microsoft.WindowsAppSDK dependency like this:

```xml
<ItemGroup>
    <PackageReference Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'" Include="Microsoft.WindowsAppSDK" Version="1.2.230313.1" />
</ItemGroup>
```

Note that you will need to do this manually by editing the project file instead of using the NuGet Package Manager UI.

**Comment by LccBuild** [comment]
> To get the fix in a .NET MAUI app, please upgrade the Microsoft.WindowsAppSDK dependency like this:
> 
> ```
> <ItemGroup>
>     <PackageReference Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'" Include="Microsoft.WindowsAppSDK" Version="1.2.230313.1" />
> </ItemGroup>
> ```

Where does that go? 
My MAUI Hybrid Blazor app project file does not seem to have an an entry containing "WindowsAppSDK"

**Comment by KieranDevvs** [comment]
> > To get the fix in a .NET MAUI app, please upgrade the Microsoft.WindowsAppSDK dependency like this:
> > ```
> > <ItemGroup>
> >     <PackageReference Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'" Include="Microsoft.WindowsAppSDK" Version="1.2.230313.1" />
> > </ItemGroup>
> > ```
> 
> Where does that go? My MAUI Hybrid Blazor app project file does not seem to have an an entry containing "WindowsAppSDK"

That's because the reference is being imported as part of the MAUI framework. By specifying a package reference in your project file, the MAUI implicit reference will be overridden.

**Comment by LccBuild** [comment]
> That's because the reference is being imported as part of the MAUI framework. By specifying a package reference in your project file, the MAUI implicit reference will be overridden.

Thanks for clarifying. 
Where do I find what WindowsAppSDK version MAUI currently uses?

**Comment by Xyncgas** [comment]
is this included inside 1.3.230502000

**Comment by arsalan16-web** [comment]
@Eilon 
can you please find out that issuse  
Referenece Ticket
https://github.com/dotnet/maui/issues/25791
https://github.com/user-attachments/assets/57319bea-f36a-451e-91bb-a5ec9d5c3ace
